lib/gst/player/gstplayer.c

```diff
@@ -1660,6 +2154,163 @@ gst_player_get_pipeline (GstPlayer * self)
   return val;
 }
 
+/**
+ * gst_player_get_media_info:
+ * @player: #GstPlayer instance
+ *
+ * Returns: (transfer full): The internal media info instance.
+ */
+GstPlayerMediaInfo*
+gst_player_get_media_info (GstPlayer *self)
+{
+  return self->priv->media_info;
+}
+
+/**
+ * gst_player_get_audio_track:
+ * @player: #GstPlayer instance
+ *
+ * Returns: (transfer full): current audio track
+ */
+GstPlayerAudioInfo*
+gst_player_get_audio_track (GstPlayer *self)
+{
+  if (!is_track_enabled (self, GST_PLAY_FLAG_AUDIO))
+    return NULL;
+
+  return (GstPlayerAudioInfo*)
+       get_current_stream (self, "current-audio", self->priv->audioinfo_list);
+}
+
+/**
+ * gst_player_get_video_track:
+ * @player: #GstPlayer instance
+ *
+ * Returns: (transfer full): current video track
+ */
+GstPlayerVideoInfo*
+gst_player_get_video_track (GstPlayer *self)
+{
+  if (!is_track_enabled (self, GST_PLAY_FLAG_VIDEO))
+    return NULL;
+
+  return (GstPlayerVideoInfo*)
+       get_current_stream (self, "current-video", self->priv->videoinfo_list);
+}
+
+/**
+ * gst_player_get_subtitle_track:
+ * @player: #GstPlayer instance
+ *
+ * Returns: (transfer full): current subtitle track
+ */
+GstPlayerSubtitleInfo*
+gst_player_get_subtitle_track (GstPlayer *self)
+{
+  if (!is_track_enabled (self, GST_PLAY_FLAG_SUBTITLE))
+    return NULL;
+
+  return (GstPlayerSubtitleInfo*)
+      get_current_stream (self, "current-text", self->priv->subtitleinfo_list);
+}
+
+/**
+ * gst_player_set_track:
+ * @player: #GstPlayer instance
+ * @s: #GstPlayerStreamInfo instance
+ */
+void
+gst_player_set_track (GstPlayer *self, GstPlayerStreamInfo *s)
+{
+  gint stream_index;
+
+  stream_index = gst_player_stream_info_get_stream_index (s);
+
+  if (GST_IS_PLAYER_VIDEO_INFO(s))
+    g_object_set (G_OBJECT(self->priv->playbin), "current-video",
+                  stream_index, NULL);
+
+  if (GST_IS_PLAYER_AUDIO_INFO(s))
+    g_object_set (G_OBJECT(self->priv->playbin), "current-audio",
+                  stream_index, NULL);
+
+  if (GST_IS_PLAYER_SUBTITLE_INFO(s))
+    g_object_set (G_OBJECT(self->priv->playbin), "current-text",
+                  stream_index, NULL);
+}
+
+/*
+ * gst_player_audio_track_enable:
+ * @player: #GstPlayer instance
+ *
+ * Enable the current audio track
+ */
+void
+gst_player_audio_track_enable (GstPlayer *self)
```

Comment:
Maybe just one method? gst_player_audio_set_enabled(self, enabled)? And the same for video/subs